pkg/controller/control.go

```diff
@@ -250,7 +253,7 @@ func (c *Controller) Expand(size int64) error {
 
 		expansionSuccess, errsNeedToBeHandled, errsForRecording := c.backend.Expand(size)
 		if errsForRecording != nil {
-			c.lastExpansionFailedAt = time.Now().UTC().Format(time.RFC3339)
+			c.lastExpansionFailedAt = time.Now().UTC().Format(time.RFC3339Nano)
```

Comment:
Why nano? for consistency?

Reply:
In the manager part, we will use `time.Now()` to do the comparison. And `time.Now()` is using nanosecond precision.

Reply:
I feel we should have a common library to use in every module, so we don't need to remember what time unit we are using.

Reply:
YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Reply:
YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Reply:
YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Reply:
then, let's create it. https://github.com/longhorn/common-libs